iron-input.html

```diff
@@ -166,7 +171,7 @@
         (event.keyCode > 32 && event.keyCode < 41)   || // page up/down, end, home, arrows
         (event.keyCode > 111 && event.keyCode < 124); // fn keys
 
-      return !(event.charCode == 0 && nonPrintable);
+      return !anyNonPrintable && !(event.charCode == 0 && mozNonPrintable);
     },
 
     _onKeypress: function(event) {
```

Comment:
Related: consider changing thisChar variable definition below, to this:

var thisChar = String.fromCharCode(event.which || event.keyCode);

See guidance on http://unixpapa.com/js/key.html.